### PR TITLE
Keycloak Authorization must block access to the public resource if a matching enforcing policy exists

### LINKED ISSUE #17164: Regression Issue: Secured endpoints are exposed when no token is provided using Keycloak-Authorization extension
## Describe the bug
Given a secured endpoint
When we call this endpoint using a invalid token
Then it returns forbidden as expected
When we call this endpoint using a valid token
Then it returns OK as expected
**When we call this endpoint with no token
Then it returns OK when it should return 404**

We are using the next Quarkus extensions:

```xml
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-keycloak-authorization</artifactId>
</dependency>
```

## To Reproduce

1. git clone `https://github.com/Sgitario/quarkus-examples`
2. cd `quarkus-examples/quarkus-keycloak-authz`
3. mvn clean verify

All the tests should pass, but this test is failing `UserResourceTest.noUser_userResource`:

```
[ERROR] Failures: 
[ERROR]   UserResourceTest.noUser_userResource:76 1 expectation failed.
Expected status code <401> but was <200>.
```

### Configuration
```properties
quarkus.oidc.auth-server-url=http://localhost:8180/auth/realms/test-realm
quarkus.oidc.client-id=test-application-client
quarkus.oidc.credentials.secret=test-application-client-secret

quarkus.keycloak.policy-enforcer.enable=true
```

### Quarkus version or git rev
It worked with `1.13.3.Final`, but started failing with `2.0.0.Alpha1` and `2.0.0.Alpha2`. Using the latest version in Main, it also fails.

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #17164

Right now, in `2.0.0.Alpha2`/main, access to a public resource with anonymous identity will succeed even if a matching `enforcing` policy is available. Instead `401` should be returned. 

CC @Sgitario

### CODE REVIEW

**Review by pedroigor — APPROVED** [review]

**Comment by sberyozkin** [comment]
@pedroigor thanks, I'd like to add one more test to verify that with a token access to a public resource will be rejected

**Comment by sberyozkin** [comment]
Hi @pedroigor and @stuartwdouglas 

So I've added two more tests - it is already tested in a few other tests, but what we have now is:
- public endpoint (no policy), no token - the endpoint is invoked

This is what the linked issue is about - before just having a policy enforcer enabled would block it with 401 - but not due to a policy requirement but because it was an anonymous request (using a token availability check). But it is not correct/consistent with how the (enabled by default) HTTP policy authorizer acts, see this [code flow test](https://github.com/quarkusio/quarkus/blob/main/integration-tests/oidc-code-flow/src/test/java/io/quarkus/it/keycloak/CodeFlowTest.java#L905) which invokes [this public endpoint](https://github.com/quarkusio/quarkus/blob/main/integration-tests/oidc-code-flow/src/main/java/io/quarkus/it/keycloak/UnprotectedResource.java)
with no http policy configured for this path in the test `application.properties`. And it was blocking valid authenticated requests if they were done by non OIDC and users had to duplicate HTTP public resource policy config with `DISABLED` enforcement mode for the same public resources.

But once the KC path policy is configured we have:

- endpoint protected by policy, no token: 401 (same as with HTTP `deny` policy for ex)
- endpoint protected by policy, token: 403 ( when token has no associated policy in KC)
- public endpoint with a policy having `enforcementMode=DISABLED`, token: endpoint is invoked

So it is all good and consistent now with the way Quarkus Sec authorization is applied to authenticated or anonymous identities.
I'm going to merge once the build passes as PR is approved unless there will be some comments

**Comment by sberyozkin** [comment]
Will keep open till tomorrow

**Comment by sberyozkin** [comment]
Added a doc note that a default tenant configuration can be used for controlling an anonymous access to the public endpoints since no tenant id is currently set by quarkus-oidc for anonymous requests

**Pull request merged**
